Packs/GigamonThreatINSIGHT/Integrations/GigamonThreatINSIGHT/GigamonThreatINSIGHT.py

```diff
@@ -9,61 +9,313 @@
 from CommonServerPython import *
 from CommonServerUserPython import *
 
-import requests
 import json
 import re
 from datetime import datetime, timedelta
 
-# Global for setting up headers
-HEADERS: Dict[str, str] = {}
 TRAINING_ACC = 'f6f6f836-8bcd-4f5d-bd61-68d303c4f634'
 MAX_DETECTIONS = 100
 DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
 
 
-''' HELPER FUNCTIONS '''
+class Client(BaseClient):
+    @staticmethod
+    def getUrl(api) -> str:
+        # set api url
+
+        url: str = ''
+        if api == 'Events':
+            url = 'https://events.icebrg.io/v2/query/'
+        elif api == 'Detections':
+            url = 'https://detections.icebrg.io/v1/'
+        elif api == 'Sensors':
+            url = 'https://sensor.icebrg.io/v1/'
+        elif api == 'Entity':
+            url = 'https://entity.icebrg.io/v1/entity/'
+
+        return url
+
+    @staticmethod
+    def getClient(api, api_key):
+        headers = {
+            'Authorization': 'IBToken ' + api_key,
+            'User-Agent': 'Cortex_Insight.v3',
+            'Content-Type': 'application/json',
+        }
+
+        match api:
```

Comment:
By using `match` here we are forcing Python3.10.  I'm not sure if that is intended or not.  It might cause problems for some people.